apps/aefate/src/aefa_fate_op.erl

```diff
@@ -1239,6 +1242,37 @@ spend(Arg0, Arg1, ES0) ->
             aefa_fate:abort({value_does_not_match_type, Other, address}, ES2)
     end.
 
+create(Arg0, Arg1, Arg2, Arg3, ES0) ->
+    ?AVAILABLE_FROM(?VM_FATE_SOPHIA_2, ES0),
+    {[ByteCode, InitArgsTypes, Value], ES1} = get_op_args([Arg1, Arg2, Arg3], ES0),
+    exit({error, todo}).
+
+clone(Arg0, Arg1, Arg2, Arg3, ES0) ->
+    ?AVAILABLE_FROM(?VM_FATE_SOPHIA_2, ES0),
+    {[Contract, InitArgsTypes, Value], ES1} = get_op_args([Arg1, Arg2, Arg3], ES0),
+    exit({error, todo}).
+
+bytecode_hash(Arg0, Arg1, ES0) ->
+    ?AVAILABLE_FROM(?VM_FATE_SOPHIA_2, ES0),
+    {[?FATE_CONTRACT(Pubkey)], ES1} = get_op_args([Arg1], ES0),
+    CodeCache = aefa_engine_state:code_cache(ES1),
+    case maps:get(Pubkey, CodeCache, void) of
+        void ->
+            case aefa_chain_api:contract_fate_code(Pubkey, APIState) of
+                {ok, ContractCode, APIState1} ->
+                    CodeCache1 = maps:put(Pubkey, ContractCode, CodeCache),
+                    ES2 = aefa_engine_state:set_code_cache(CodeCache1, ES1),
+                    ES3 = aefa_engine_state:set_chain_api(APIState1, ES2),
+                    Hashed = aeb_fate_data:make_hash(aec_hash:hash(fate_bytecode, ContractCode)),
+                    write(Arg0, Hashed)
+                error ->
+                    abort({trying_to_get_contract_hash, Pubkey}, ES1)
+            end;
+        ContractCode ->
+            Hashed = aeb_fate_data:make_hash(aec_hash:hash(fate_bytecode, ContractCode)),
```

Comment:
It this a hash over the code or is this the hash over bytecode? Also is this the hash over the serialized fcode or is it over the deserialized version of the code?

Reply:
Bytecode, I believe serialized. Must check tho